### DISCUSSION TITLE
Performance problem with hot reaload in nuxt 2 and tailwind 3

### DISCUSSION THREAD

**Discussion by axelhzf** [tailwindlabs/tailwindcss#8055] [Help] [answered]
I am working on a nuxt 2 application, and after migrating to tailwind 3, the times build times on dev mode have increased significantly. A single component change:

- tailwind 2 (jit mode) : ~2s
- tailwind 3 : ~10s

I follow the [official instructions](https://tailwindcss.com/docs/guides/nuxtjs) to install tailwind.

I tried to debug the problem and I am seeing that tailwind is processing many files. I suspect that is being called for the modified file **AND for each component that contains an `@apply`**

![image](https://user-images.githubusercontent.com/175627/161971681-5e130a8d-b7f0-4a71-9a8a-296a743a45e6.png)

On the other hand, using tailwind 2, it is only being called for the modified file 

![image](https://user-images.githubusercontent.com/175627/161971962-aea4006a-c87c-4796-a5b1-f2f960908a91.png)

**Comment by axelhzf** [comment]
I was reading https://tailwindcss.com/docs/functions-and-directives#using-apply-with-per-component-css and it says

> That means if you have 10 components that each have a <style> block, Tailwind is being run 10 separate times, and each run has zero knowledge about the other runs. Because of this, when you try to @apply card in Card.svelte it fails, because Tailwind has no idea that the card class exists since Svelte processed Card.svelte and main.css in total isolation from each other.

Maybe this is the problem and tailwind a different tailwind instance is running for each component. In this case, the tracking content for these instances should be only the modified file and not every file defined on the tailwind config.

**Comment by axelhzf** [comment]
I created a repo with the minimal setup here https://github.com/axelhzf/nuxt-tailwind-performance

Running the project with the debug enabled (`DEBUG=true yarn dev`). This is what I see as soon as I make a change on the index page

```
Source path: /Users/axelhernandezferrera/dev/nuxt-tailwind-performance/components/ComponentWithApply.vue
Finding changed files: 1.361ms                                                     16:23:59
JIT TOTAL: 2.493ms                                                                 16:23:59
                                                                                   16:23:59

                                                                                   16:23:59

Source path: /Users/axelhernandezferrera/dev/nuxt-tailwind-performance/components/ComponentWithApply2.vue
Finding changed files: 0.768ms                                                     16:23:59
JIT TOTAL: 1.671ms                                                                 16:23:59
                                                                                   16:23:59

                                                                                   16:23:59

Source path: /Users/axelhernandezferrera/dev/nuxt-tailwind-performance/assets/css/main.css
Finding changed files: 0.784ms                                                     16:23:59
Reading changed files: 0.005ms                                                     16:23:59
Generate rules: 0.067ms                                                            16:23:59
Build stylesheet: 0.003ms                                                          16:23:59
Potential classes:  1                                                              16:23:59
Active contexts:  3                                                                16:23:59
JIT TOTAL: 4.115ms
```

**Comment by axelhzf** [comment, marked as answer]
Actually, the problem is already solved in https://github.com/tailwindlabs/tailwindcss/pull/7524

**Comment by ztickm** [comment]
I'm still having this same problem on with Tailwind 3.2.1 and Nuxt 2, anyone else still has the problem or have an idea why?
